For resource_pools, only bring back usable Resource Pools

kbrock opened the pull request:
Part of the `/ems_infra/:id?display=ems_folders series` ( #12002). Extracted from 
#12072 

**before:**
All resource pools are fetched from the db. Only the ones at a certain depth are kept.

**after:**
Only resource pools at the preferred depth are retrieved from the db..

|        ms |queries | query (ms) |     rows |`comments`
|       ---:|  ---:|      ---:|      ---:| ---
|  22,688.1 |  288 | 21,074.3 |    1,403 |master
|   4,562.3 |  287 |  3,004.4 |      233 |after
|                 | 0%    |                | 83% |

Once you apply other PRs, the ms and query ms will probably be comperable. The number of rows brought back and therefore the number of records instantiated is the real savings - 83%.

Fryguy commented:
LGTM but tests failing.

Fryguy commented on app/models/mixins/relationship_mixin.rb:
```diff
@@ -390,6 +390,22 @@ def subtree_arranged(*args)
+  def child_and_grandchild_rels(*args)
+    options = args.extract_options!
+    rels = relationships.inject(Relationship.none) do |stmt, r|
+      stmt.or(Relationship.where(r.child_and_grand_conditions))
```
tests failing because `child_and_grand_conditions` should be `child_and_grandchild_conditions`

Fryguy commented:
Also you got EmsCluster from an F to a D with this PR..  cc @jrafanie 🎉

Fryguy commented:
@kbrock There's a rubocop thing here that you might want to fix.

kbrock commented:
Odd. Ran tests locally and they were passing. (but the error looked like it didn't pick up the changes)
Fixed cop, rebased, and hoping they will pass this time

chessbyte commented:
Euwe Backport details:
```sh
$ git log -1
commit 46f4661eb3346660eb2092b140b3e2ad9cc86a10
Author: Jason Frey <fryguy9@gmail.com>
Date:   Thu Oct 27 13:29:27 2016 -0400

    Merge pull request #12215 from kbrock/grandchild_rels
    
    For resource_pools, only bring back usable Resource Pools
    (cherry picked from commit 585c7f1a6261c1879be09c6fcb8a1f5a489a10a3)
```
